packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -96,7 +100,8 @@ export const useGridSelection = (apiRef: GridApiRef): void => {
       apiRef.current.publishEvent(GRID_ROW_SELECTED, rowSelectedParam);
       apiRef.current.publishEvent(GRID_SELECTION_CHANGED, selectionChangeParam);
     },
-    [apiRef, logger, options.checkboxSelection, forceUpdate, setGridState],
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
What's missing? Cases where ignoring the warning is welcome are rare. At the very least, I think that we should document it.

x2

Reply:
ESLint was complaining that `options` should a dependency. I changed to destructure the options like recommended [here](https://github.com/facebook/react/issues/15924#issuecomment-521253636).

![image](https://user-images.githubusercontent.com/42154031/118161181-add97080-b3f5-11eb-94e9-2b753910238a.png)

Reply:
Ahhh, I ever wondered why the eslint rule was behaving this way.